pkg/api/vm/handler.go

```diff
@@ -397,6 +429,85 @@ func (h *vmActionHandler) abortMigration(namespace, name string) error {
 	return nil
 }
 
+func (h *vmActionHandler) findMigratableNodes(rw http.ResponseWriter, namespace, name string) error {
+	vmi, err := h.vmiCache.Get(namespace, name)
+	if err != nil {
+		return err
+	}
+	nodes, err := h.findMigratableNodesByVMI(vmi)
+	if err != nil {
+		return err
+	}
+	resp := GetMigratableNodesOutput{
+		Nodes: nodes,
+	}
+
+	util.ResponseOKWithBody(rw, resp)
+	return nil
+}
+
+func (h *vmActionHandler) findMigratableNodesByVMI(vmi *kubevirtv1.VirtualMachineInstance) ([]string, error) {
+	nodeSelector := labels.Everything()
+
+	// get node selector from network
+	vmNetworks := vmi.Spec.Networks
+	for _, network := range vmNetworks {
+		requirement, err := h.getNodeSelectorRequirementFromNetwork(vmi.Namespace, network)
+		if err != nil {
+			return nil, err
+		}
+		if requirement != nil {
+			nodeSelector = nodeSelector.Add(*requirement)
+		}
+	}
+	nodes, err := h.nodeCache.List(nodeSelector)
+	if err != nil {
+		return nil, err
+	}
+	if len(nodes) == 0 {
+		return nil, nil
+	}
```

Comment:
prefer to define the `migratableNodes` var on the top and return it here instead.